tdms/src/arrays.cpp

```diff
@@ -237,3 +237,28 @@ void DTilde::initialise(const mxArray *ptr, int n_rows, int n_cols) {
   y = component_in(ptr, "Dy_tilde", n_rows, n_cols);
   n_det_modes = mxGetDimensions(ptr_to_nd_array_in(ptr, 3, "Dx_tilde", "D_tilde"))[0];
 }
+
+Tensor3D<double> IncidentField::component_in(const mxArray *ptr, const std::string &name){
+
+  if (mxIsEmpty(mxGetField(ptr, 0, name.c_str()))) {
+    cerr << name+" not present" << endl;
+    return {};
+  }
+
+  auto element = ptr_to_nd_array_in(ptr, 3, name, "tdfield");
+  auto dims = mxGetDimensions(element);
+  int N = dims[0], M = dims[1], O = dims[2];
+  auto field = Tensor3D<double>(cast_matlab_3D_array(mxGetPr(element), N, M, O), O, M, N);
+  field.is_matlab_initialised = true;
+
+  cerr << "Got tdfield, dims=("+to_string(N)+","+to_string(M)+","+to_string(O)+")" << endl;
```

Comment:
Do you know why we're outputting to `cerr`  vs. `cout` (since this is more like information)?
(Peter was doing it [here](https://github.com/t-young31/ifdtd/pull/13/commits/2625cbafe6f6e419c74a2207491c7927153307ef#diff-1b4f29097ad84ad119a5daf882ffa97a954158d46e380098653df75a1e08e901L537))

Reply:
not a clue – kept it for consistency. seems like it would be better as log.info() anyway..(?)

Reply:
Yeah, right. Once UCL/TDMS#83 is merged, someone (probably me) can go trhough and sane-ify the printout.